icicle_v3/tests/test_field_api.cpp

```diff
@@ -269,30 +293,32 @@ TYPED_TEST(FieldApiTest, ntt)
     ext.set(CUDA_NTT_FAST_TWIDDLES_MODE, true);
     init_domain_config.ext = &ext;
 
-    ICICLE_CHECK(ntt_init_domain(scalar_t::omega(logn), init_domain_config));
-
     auto config = default_ntt_config<scalar_t>();
-
-    TypeParam *d_in, *d_out;
-    icicle_malloc_async((void**)&d_in, N * sizeof(TypeParam), config.stream);
-    icicle_malloc_async((void**)&d_out, N * sizeof(TypeParam), config.stream);
-    icicle_copy_to_device_async(d_in, scalars.get(), N * sizeof(TypeParam), config.stream);
-
-    config.ordering = Ordering::kNN;
     config.stream = stream;
+    config.coset_gen = coset_gen;
+    config.batch_size = batch_size;       // default: 1
+    config.columns_batch = columns_batch; // default: false
+    config.ordering = ordering;           // default: kNN
     config.are_inputs_on_device = true;
     config.are_outputs_on_device = true;
+    config.is_async = true; // default: false
+    ICICLE_CHECK(ntt_init_domain(scalar_t::omega(17), init_domain_config));
```

Comment:
define the log_ntt_domain_size as a variable (like ITERS on top) and use that instead of hardcoded 17.